src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,392 @@
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            return endPoints;
+        }
+
+        public override IcmpV4Statistics GetIcmpV4Statistics()
+        {
+            return LinuxIcmpV4Statistics.CreateIcmpV4Statistics();
+        }
+
+        public override IcmpV6Statistics GetIcmpV6Statistics()
+        {
+            return LinuxIcmpV6Statistics.CreateIcmpV6Statistics();
+        }
+
+        public override IPGlobalStatistics GetIPv4GlobalStatistics()
+        {
+            return LinuxIPGlobalStatistics.CreateIPv4Statistics();
+        }
+
+        public override IPGlobalStatistics GetIPv6GlobalStatistics()
+        {
+            return LinuxIPGlobalStatistics.CreateIPv6Statistics();
+        }
+
+        public override TcpStatistics GetTcpIPv4Statistics()
+        {
+            return new LinuxTcpStatistics(ipv6:false);
+        }
+
+        public override TcpStatistics GetTcpIPv6Statistics()
+        {
+            return new LinuxTcpStatistics(ipv6:true);
+        }
+
+        public override UdpStatistics GetUdpIPv4Statistics()
+        {
+            return LinuxUdpStatistics.CreateUdpIPv4Statistics();
+        }
+
+        public override UdpStatistics GetUdpIPv6Statistics()
+        {
+            return LinuxUdpStatistics.CreateUdpIPv6Statistics();
+        }
+
+        public override UnicastIPAddressInformationCollection GetUnicastAddresses()
+        {
+            UnicastIPAddressInformationCollection collection = new UnicastIPAddressInformationCollection();
+            foreach (UnicastIPAddressInformation info in
+                LinuxNetworkInterface.GetLinuxNetworkInterfaces().SelectMany(lni => lni.GetIPProperties().UnicastAddresses))
+            {
+                // PERF: Use Interop.Sys.EnumerateInterfaceAddresses directly here.
+                collection.InternalAdd(info);
+            }
+
+            return collection;
+        }
+
+        public override IAsyncResult BeginGetUnicastAddresses(AsyncCallback callback, object state)
+        {
+            Task<UnicastIPAddressInformationCollection> t = GetUnicastAddressesAsync();
+            return TaskToApm.Begin(t, callback, state);
+        }
+
+        public override UnicastIPAddressInformationCollection EndGetUnicastAddresses(IAsyncResult asyncResult)
+        {
+            return TaskToApm.End<UnicastIPAddressInformationCollection>(asyncResult);
+        }
+
+        public override Task<UnicastIPAddressInformationCollection> GetUnicastAddressesAsync()
+        {
+            return Task.Run((Func<UnicastIPAddressInformationCollection>)GetUnicastAddresses);
```

Comment:
#4044: We are removing non-async versions of this API. 
@mellinoe It would be nice to switch the File I/O access to async to avoid creating a task and blocking a thread for I/O purposes only. (Maybe we can track this with a TODO?)